src/verilog/verilog_synthesis.cpp

```diff
@@ -405,6 +405,69 @@ exprt verilog_synthesist::synth_expr(exprt expr, symbol_statet symbol_state)
 
 /*******************************************************************\
 
+Function: verilog_synthesist::synth_lhs_expr
+
+  Inputs:
+
+ Outputs:
+
+ Purpose:
+
+\*******************************************************************/
+
+exprt verilog_synthesist::synth_lhs_expr(exprt expr)
+{
+  if(expr.id() == ID_symbol)
+  {
+    return expr; // leave as is
+  }
+  else if(expr.id() == ID_concatenation)
+  {
+    for(auto &op : expr.operands())
+      op = synth_lhs_expr(op);
+
+    return expr;
+  }
```

Comment:
What are the semantics of a concatenation as an lvalue?

Reply:
Piece-wise assignment to each component.